### ISSUE TITLE
RouteConfigLoadCancel, RouteConfigLoadError

### ISSUE THREAD

**Issue by edwardboyle** [angular/angular#29894]
# 🚀 RouteConfigLoadCancel, RouteConfigLoadError

### Description
We Like this:
```typescript
 if (event instanceof RouteConfigLoadStart) {
  this.isLoading = true;
  } else if (event instanceof RouteConfigLoadEnd) {
  this.isLoading = false;
}
```
_Many developers are making progress indicators based on this event with dozens of blog post based on using these events._ 

However, `RouteConfigLoadEnd` is not guaranteed to fire. 

### Describe the solution you'd like
`RouteConfigLoadCancel` `RouteConfigLoadError` 

### Alternative
```typescript
export enum RouteConfigLoadEndState{
 Success,
 Cancel,
 Error
}
```

```typescript
class RouteConfigLoadEnd{
  //...
  state: RouteConfigLoadEndState
}
```

**Comment by humana-fragilitas** [comment]
Lazy loaded modules are compiled into Webpack chunks fetched at runtime via script tags; the only hint to any possible error you get when a lazy loaded route fails is a [ChunkLoadError](https://github.com/webpack/webpack/issues/9611); either success or failure: no cancel option. Do you think that the sole **RouteConfigLoadError** event would anyway be a nifty feature to add?

**Comment by pgammans** [comment]
I meet this issues while writing a test for #36760 and was looking to raise/fix this issues.

Adding a error handler to the call to returned observable from loadModuleFactory would allow the event to be sent at a similar point to where the successful RouteConfigLoadEnd is sent.

It should be noted (in reference to the original usage description )
* That RouteConfigLoadEnd is sent only when the 'load' operation is complete it doesn't signify the module is active  Ie #36605 could still cause a failure but a RouteConfigLoadStart and RouteConfigLoadEnd would have still be sent be.
* There is the possibility of multiple RouteConfigLoadEnd set until #22842 is fixed


1: Do the angular team want a custom event similar to @jasonaden request for #16308?
2: If this ticket waits till after #36654 is merged, then the error handler could be added later ensuring a RouteConfigReady or RouteConfigFailure event was sent.

**Comment by alxhub** [comment]
Closing as a spiritual duplicate of #31800 (would be solved by this issue)
